test/functional/test_sites

```diff
@@ -0,0 +1,2 @@
+test_site
+test_site_algolia_plugin
```

Comment:
Screenshot of the current docs:
<img width="715" alt="Screenshot 2019-04-10 at 11 32 31 PM" src="https://user-images.githubusercontent.com/9073504/55892510-03c10600-5be9-11e9-91d3-819a414f05dd.png">

Given that we no longer have one ("the") test site, it would be great to also update the details of this new testing approach in our developer guide (in the `Running tests` and `Updating tests` sections). In particular, mentioning that whenever a new test site is added, its folder name will need to be added to this file.